### PR TITLE
Add postprocess module containing IPR class and add calc_IPR function

### PR DESCRIPTION
**Pull request by timcallow**
`postprocess` module created which will contain routines which perform miscellaneous and inexpensive tasks on data generated by an SCF calculation.

The first class to be add is the `IPR` class which calculates the inverse participation ratio for the orbitals, and quantities derived from it.

### CODE REVIEW

**Comment by DanielKotik** [comment]
To me it looks like the last commit 24156b6def33e62191cdfbc2e1157f014e00487f is obsolete and introduces the problems. If the actual PR consists of only these commits

- 8ac42b20f8f7e67782167ba064a370fa899439e8
- 0fab930c26f103a9dc50757a5f18b1a28bf6cb19

my advice would be to delete the "erroneously" generated "merge" commit 24156b6def33e62191cdfbc2e1157f014e00487f  locally (we can discuss how such commits are avoided and what are good practices) and force push `inverse_participation_ratio` again. This would result in a clean history. I must, however, note the following: force push into a public repository which is the original repo of a project can cause side effects. In this case, problems would arise if someone in their local copy of the atoMEC repo had already checked out the branch `inverse_participation_ratio`. After you have rewritten the history (this is what a force push will do) the upstream branch diverges from its local copy. It's almost always better to work with a fork - even for the main developers. Force pushing to a public fork does not affect anyone except you (and people that my have cloned your fork - but who would do that?!). So please consider making PRs from a fork in the future :wink:.

These are the relevant commands
```sh
git checkout inverse_participation_ratio
git reset --hard HEAD^
git push -f
```

**Comment by timcallow** [comment]
Thanks for the advice, I have done the force push as suggested. From now on I promise to use my fork!

File: atoMEC/postprocess.py [hunk 1]
```diff
@@ -1,78 +1,65 @@
...
+    lmax = np.shape(eigfuncs)[1]
+    nmax = np.shape(eigfuncs)[2]
+
+    IPR_mat = np.zeros((spindims, lmax, nmax))
+
+    # compute the IPR matrix
+    # FIXME: add spherical harmonic term
+    for i in range(spindims):
+        for l in range(lmax):
+            for n in range(nmax):
```
Comment (DanielKotik) [inline atoMEC/postprocess.py:59]: These nested loops can cause performance bottlenecks and are potential candidates for vectorization (looping over numpy arrays should be avoided). For now, I've no idea if this is already performance critical or not....

Comment (timcallow) [reply atoMEC/postprocess.py:59]: Mmm, luckily the product `nmax * lmax * spindims` is never that large (max few thousand) and the operations inside the loop are not expensive. (During tests on my laptop, things always ran basically instantaneously). I would say it's ok for now.

**Review by DanielKotik — APPROVED** [review]

**Pull request merged**